src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2147,6 +2121,26 @@ unsafe private static void ThreadSafeCopy(char* sourcePtr, char[] destination, i
                 }
             }
         }
+
+        private static unsafe void ThreadSafeCopy(char[] source, int sourceIndex, char* destinationPtr, int destinationIndex, int count)
```

Comment:
It appears that this overload was only used in the CoreRt specific stuff, so I moved it there.